operator/internal/decommissioning/statefulset_decommissioner_test.go

```diff
@@ -0,0 +1,396 @@
+	})
+	s.Require().NoError(s.client.Update(s.ctx, &node))
+}
+
+func (s *StatefulSetDecommissionerSuite) untaintNode(name string) {
+	var node corev1.Node
+	s.Require().NoError(s.client.Get(s.ctx, types.NamespacedName{Name: name}, &node))
+	node.Spec.Taints = functional.Filter(node.Spec.Taints, func(taint corev1.Taint) bool {
+		return taint.Key != "decommission-test"
+	})
+	s.Require().NoError(s.client.Update(s.ctx, &node))
+}
+
+func (s *StatefulSetDecommissionerSuite) SetupSuite() {
+	t := s.T()
+
+	scheme := runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	log := testr.New(t).V(10)
+
+	s.ctx = context.Background()
+	s.env = testenv.New(t, testenv.Options{
+		// We need our own cluster for these tests since we need additional
+		// agents. Otherwise we can just turn up the default... but we'll
+		// need a different cluster to manipulate for node cleanup anyway.
+		Name:   "decommissioning",
+		Agents: 5,
+		Scheme: scheme,
+		Logger: log,
+	})
+
+	s.client = s.env.Client()
+
+	s.env.SetupManager(s.setupRBAC(), func(mgr ctrl.Manager) error {
+		helmClient, err := helm.New(helm.Options{
+			KubeConfig: mgr.GetConfig(),
+		})
+		if err != nil {
+			return err
+		}
+		if err := helmClient.RepoAdd(s.ctx, "redpandadata", "https://charts.redpanda.com"); err != nil {
+			return err
+		}
+
+		s.helm = helmClient
+		dialer := kube.NewPodDialer(mgr.GetConfig())
+		s.clientFactory = internalclient.NewFactory(mgr.GetConfig(), mgr.GetClient()).WithDialer(dialer.DialContext)
+
+		options := []decommissioning.Option{
+			// override this so we can dial directly to our Redpanda pods
+			decommissioning.WithFactory(s.clientFactory),
+			// set these low so that we don't have to wait forever in the test
+			// these settings should give about a 5-10 second window before
+			// actually running a decommission
+			decommissioning.WithDelayedCacheInterval(5 * time.Second),
+			decommissioning.WithDelayedCacheMaxCount(2),
+			decommissioning.WithRequeueTimeout(2 * time.Second),
+		}
+		decommissioner := decommissioning.NewStatefulSetDecommissioner(mgr, decommissioning.NewHelmFetcher(mgr), options...)
+		if err := decommissioner.SetupWithManager(mgr); err != nil {
+			return err
+		}
+
+		return nil
+	})
+}
+
+type chart struct {
+	name    string
+	version string
+	release helm.Release
+	values  map[string]any
+}
+
+func (s *StatefulSetDecommissionerSuite) installChart(name, version string, overrides map[string]any) *chart {
+	values := map[string]any{
+		"statefulset": map[string]any{
+			"replicas": 1,
+		},
+		"console": map[string]any{
+			"enabled": false,
+		},
+		"external": map[string]any{
+			"enabled": false,
+		},
+		"image": map[string]any{
+			"repository": "redpandadata/redpanda-unstable",
+			"tag":        "v24.3.1-rc8",
+		},
+	}
+
+	if overrides != nil {
+		values = functional.MergeMaps(values, overrides)
+	}
+
+	release, err := s.helm.Install(s.ctx, "redpandadata/redpanda", helm.InstallOptions{
```

Comment:
Not a problem: What would be better option as we have charts in redpanda-operator repository, so it could just reference path within this repo.